mitsubishiheatpump + neopixelbus = compile error :(

Trying to get mitsubishiheatpump and neopixelbus working together in one esphome esp8266.

Independently, they both work great on the same device - however, when I try and include them both it spits out  "error: call of overloaded 'micros()' is ambiguous"

Relevant Esphome v2022.12.8 yaml:
`light:
  - platform: neopixelbus
    type: grb
    pin: GPIO15
    num_leds: 92
    name: "Aircon Mood Lights"
    variant: ws2812x
    
climate:
  - platform: mitsubishi_heatpump
    name: "Living Room Aircon"
    hardware_uart: UART0`

Full output when compiling:
`HARDWARE: ESP8266 80MHz, 80KB RAM, 1MB Flash
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
Dependency Graph
|-- ESPAsyncTCP-esphome @ 1.2.3
|-- ESPAsyncWebServer-esphome @ 2.1.0
|   |-- ESPAsyncTCP-esphome @ 1.2.3
|   |-- Hash @ 1.0
|   |-- ESP8266WiFi @ 1.0
|-- DNSServer @ 1.1.1
|-- ESP8266WiFi @ 1.0
|-- ESP8266mDNS @ 1.2
|-- noise-c @ 0.1.4
|   |-- libsodium @ 1.10018.1
|-- AsyncMqttClient-esphome @ 0.8.6
|   |-- ESPAsyncTCP-esphome @ 1.2.3
|-- ArduinoJson @ 6.18.5
|-- NeoPixelBus @ 2.6.9
|   |-- SPI @ 1.0
|-- HeatPump @ 1.0.0+sha.1e7e596
Compiling .pioenvs/living-room-aircon/src/esphome/components/mitsubishi_heatpump/espmhp.cpp.o
Compiling .pioenvs/living-room-aircon/src/esphome/components/web_server/list_entities.cpp.o
Compiling .pioenvs/living-room-aircon/src/esphome/components/web_server/web_server.cpp.o
Compiling .pioenvs/living-room-aircon/src/esphome/components/web_server_base/web_server_base.cpp.o
Compiling .pioenvs/living-room-aircon/src/esphome/components/wifi/wifi_component.cpp.o
Compiling .pioenvs/living-room-aircon/src/esphome/components/wifi/wifi_component_esp32_arduino.cpp.o
In file included from .piolibdeps/living-room-aircon/NeoPixelBus/src/NeoPixelBus.h:96,
                 from src/esphome/components/neopixelbus/neopixelbus_light.h:12,
                 from src/esphome.h:73,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/Ws2801GenericMethod.h: In member function 'bool Ws2801MethodBase<T_TWOWIRE>::IsReadyToUpdate() const':
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/Ws2801GenericMethod.h:64:33: error: call of overloaded 'micros()' is ambiguous
   64 |         uint32_t delta = micros() - _endTime;
      |                                 ^
In file included from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/PolledTimeout.h:29,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Stream.h:28,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Udp.h:38,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/ESP8266WiFi/src/WiFiUdp.h:25,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/DNSServer/src/DNSServer.h:3,
                 from src/esphome/components/captive_portal/captive_portal.h:6,
                 from src/esphome.h:19,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
/config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/core_esp8266_features.h:103:15: note: candidate: 'long unsigned int micros()'
  103 | unsigned long micros(void);
      |               ^~~~~~
In file included from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:4,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
src/esphome/core/hal.h:38:10: note: candidate: 'uint32_t esphome::micros()'
   38 | uint32_t micros();
      |          ^~~~~~
In file included from .piolibdeps/living-room-aircon/NeoPixelBus/src/NeoPixelBus.h:96,
                 from src/esphome/components/neopixelbus/neopixelbus_light.h:12,
                 from src/esphome.h:73,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/Ws2801GenericMethod.h: In member function 'void Ws2801MethodBase<T_TWOWIRE>::Initialize()':
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/Ws2801GenericMethod.h:82:27: error: call of overloaded 'micros()' is ambiguous
   82 |         _endTime = micros();
      |                           ^
In file included from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/PolledTimeout.h:29,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Stream.h:28,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Udp.h:38,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/ESP8266WiFi/src/WiFiUdp.h:25,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/DNSServer/src/DNSServer.h:3,
                 from src/esphome/components/captive_portal/captive_portal.h:6,
                 from src/esphome.h:19,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
/config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/core_esp8266_features.h:103:15: note: candidate: 'long unsigned int micros()'
  103 | unsigned long micros(void);
      |               ^~~~~~
In file included from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:4,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
src/esphome/core/hal.h:38:10: note: candidate: 'uint32_t esphome::micros()'
   38 | uint32_t micros();
      |          ^~~~~~
In file included from .piolibdeps/living-room-aircon/NeoPixelBus/src/NeoPixelBus.h:96,
                 from src/esphome/components/neopixelbus/neopixelbus_light.h:12,
                 from src/esphome.h:73,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/Ws2801GenericMethod.h: In member function 'void Ws2801MethodBase<T_TWOWIRE>::Update(bool)':
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/Ws2801GenericMethod.h:90:19: error: call of overloaded 'yield()' is ambiguous
   90 |             yield(); // allows for system yield if needed
      |                   ^
In file included from .piolibdeps/living-room-aircon/ESPAsyncWebServer-esphome/src/ESPAsyncWebServer.h:24,
                 from src/esphome/components/web_server_base/web_server_base.h:11,
                 from src/esphome/components/captive_portal/captive_portal.h:10,
                 from src/esphome.h:19,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
/config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:193:6: note: candidate: 'void yield()'
  193 | void yield(void);
      |      ^~~~~
In file included from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:4,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
src/esphome/core/hal.h:36:6: note: candidate: 'void esphome::yield()'
   36 | void yield();
      |      ^~~~~
In file included from .piolibdeps/living-room-aircon/NeoPixelBus/src/NeoPixelBus.h:96,
                 from src/esphome/components/neopixelbus/neopixelbus_light.h:12,
                 from src/esphome.h:73,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/Ws2801GenericMethod.h:102:27: error: call of overloaded 'micros()' is ambiguous
  102 |         _endTime = micros();
      |                           ^
In file included from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/PolledTimeout.h:29,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Stream.h:28,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Udp.h:38,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/ESP8266WiFi/src/WiFiUdp.h:25,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/DNSServer/src/DNSServer.h:3,
                 from src/esphome/components/captive_portal/captive_portal.h:6,
                 from src/esphome.h:19,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
/config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/core_esp8266_features.h:103:15: note: candidate: 'long unsigned int micros()'
  103 | unsigned long micros(void);
      |               ^~~~~~
In file included from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:4,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
src/esphome/core/hal.h:38:10: note: candidate: 'uint32_t esphome::micros()'
   38 | uint32_t micros();
      |          ^~~~~~
In file included from .piolibdeps/living-room-aircon/NeoPixelBus/src/NeoPixelBus.h:101,
                 from src/esphome/components/neopixelbus/neopixelbus_light.h:12,
                 from src/esphome.h:73,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266DmaMethod.h: In destructor 'NeoEsp8266DmaMethodBase<T_SPEED>::~NeoEsp8266DmaMethodBase()':
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266DmaMethod.h:343:19: error: call of overloaded 'yield()' is ambiguous
  343 |             yield();
      |                   ^
In file included from .piolibdeps/living-room-aircon/ESPAsyncWebServer-esphome/src/ESPAsyncWebServer.h:24,
                 from src/esphome/components/web_server_base/web_server_base.h:11,
                 from src/esphome/components/captive_portal/captive_portal.h:10,
                 from src/esphome.h:19,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
/config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:193:6: note: candidate: 'void yield()'
  193 | void yield(void);
      |      ^~~~~
In file included from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:4,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
src/esphome/core/hal.h:36:6: note: candidate: 'void esphome::yield()'
   36 | void yield();
      |      ^~~~~
In file included from .piolibdeps/living-room-aircon/NeoPixelBus/src/NeoPixelBus.h:101,
                 from src/esphome/components/neopixelbus/neopixelbus_light.h:12,
                 from src/esphome.h:73,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266DmaMethod.h:348:32: error: call of overloaded 'micros()' is ambiguous
  348 |         uint32_t time = micros();
      |                                ^
In file included from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/PolledTimeout.h:29,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Stream.h:28,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Udp.h:38,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/ESP8266WiFi/src/WiFiUdp.h:25,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/DNSServer/src/DNSServer.h:3,
                 from src/esphome/components/captive_portal/captive_portal.h:6,
                 from src/esphome.h:19,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
/config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/core_esp8266_features.h:103:15: note: candidate: 'long unsigned int micros()'
  103 | unsigned long micros(void);
      |               ^~~~~~
In file included from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:4,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
src/esphome/core/hal.h:38:10: note: candidate: 'uint32_t esphome::micros()'
   38 | uint32_t micros();
      |          ^~~~~~
In file included from .piolibdeps/living-room-aircon/NeoPixelBus/src/NeoPixelBus.h:101,
                 from src/esphome/components/neopixelbus/neopixelbus_light.h:12,
                 from src/esphome.h:73,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266DmaMethod.h:349:24: error: call of overloaded 'micros()' is ambiguous
  349 |         while ((micros() - time) < ((getPixelTime() + T_SPEED::ResetTimeUs) * waits))
      |                        ^
In file included from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/PolledTimeout.h:29,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Stream.h:28,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Udp.h:38,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/ESP8266WiFi/src/WiFiUdp.h:25,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/DNSServer/src/DNSServer.h:3,
                 from src/esphome/components/captive_portal/captive_portal.h:6,
                 from src/esphome.h:19,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
/config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/core_esp8266_features.h:103:15: note: candidate: 'long unsigned int micros()'
  103 | unsigned long micros(void);
      |               ^~~~~~
In file included from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:4,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
src/esphome/core/hal.h:38:10: note: candidate: 'uint32_t esphome::micros()'
   38 | uint32_t micros();
      |          ^~~~~~
In file included from .piolibdeps/living-room-aircon/NeoPixelBus/src/NeoPixelBus.h:101,
                 from src/esphome/components/neopixelbus/neopixelbus_light.h:12,
                 from src/esphome.h:73,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266DmaMethod.h:351:19: error: call of overloaded 'yield()' is ambiguous
  351 |             yield();
      |                   ^
In file included from .piolibdeps/living-room-aircon/ESPAsyncWebServer-esphome/src/ESPAsyncWebServer.h:24,
                 from src/esphome/components/web_server_base/web_server_base.h:11,
                 from src/esphome/components/captive_portal/captive_portal.h:10,
                 from src/esphome.h:19,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
/config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:193:6: note: candidate: 'void yield()'
  193 | void yield(void);
      |      ^~~~~
In file included from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:4,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
src/esphome/core/hal.h:36:6: note: candidate: 'void esphome::yield()'
   36 | void yield();
      |      ^~~~~
In file included from .piolibdeps/living-room-aircon/NeoPixelBus/src/NeoPixelBus.h:101,
                 from src/esphome/components/neopixelbus/neopixelbus_light.h:12,
                 from src/esphome.h:73,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266DmaMethod.h: In member function 'void NeoEsp8266DmaMethodBase<T_SPEED>::Update(bool)':
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266DmaMethod.h:484:19: error: call of overloaded 'yield()' is ambiguous
  484 |             yield();
      |                   ^
In file included from .piolibdeps/living-room-aircon/ESPAsyncWebServer-esphome/src/ESPAsyncWebServer.h:24,
                 from src/esphome/components/web_server_base/web_server_base.h:11,
                 from src/esphome/components/captive_portal/captive_portal.h:10,
                 from src/esphome.h:19,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
/config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:193:6: note: candidate: 'void yield()'
  193 | void yield(void);
      |      ^~~~~
In file included from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:4,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
src/esphome/core/hal.h:36:6: note: candidate: 'void esphome::yield()'
   36 | void yield();
      |      ^~~~~
In file included from .piolibdeps/living-room-aircon/NeoPixelBus/src/NeoPixelBus.h:102,
                 from src/esphome/components/neopixelbus/neopixelbus_light.h:12,
                 from src/esphome.h:73,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266UartMethod.h: In destructor 'NeoEsp8266Uart<T_UARTFEATURE, T_UARTCONTEXT>::~NeoEsp8266Uart()':
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266UartMethod.h:182:19: error: call of overloaded 'yield()' is ambiguous
  182 |             yield();
      |                   ^
In file included from .piolibdeps/living-room-aircon/ESPAsyncWebServer-esphome/src/ESPAsyncWebServer.h:24,
                 from src/esphome/components/web_server_base/web_server_base.h:11,
                 from src/esphome/components/captive_portal/captive_portal.h:10,
                 from src/esphome.h:19,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
/config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:193:6: note: candidate: 'void yield()'
  193 | void yield(void);
      |      ^~~~~
In file included from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:4,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
src/esphome/core/hal.h:36:6: note: candidate: 'void esphome::yield()'
   36 | void yield();
      |      ^~~~~
In file included from .piolibdeps/living-room-aircon/NeoPixelBus/src/NeoPixelBus.h:102,
                 from src/esphome/components/neopixelbus/neopixelbus_light.h:12,
                 from src/esphome.h:73,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266UartMethod.h: In member function 'void NeoEsp8266Uart<T_UARTFEATURE, T_UARTCONTEXT>::UpdateUart(bool)':
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266UartMethod.h:197:29: error: call of overloaded 'micros()' is ambiguous
  197 |         _startTime = micros();
      |                             ^
In file included from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/PolledTimeout.h:29,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Stream.h:28,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Udp.h:38,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/ESP8266WiFi/src/WiFiUdp.h:25,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/DNSServer/src/DNSServer.h:3,
                 from src/esphome/components/captive_portal/captive_portal.h:6,
                 from src/esphome.h:19,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
/config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/core_esp8266_features.h:103:15: note: candidate: 'long unsigned int micros()'
  103 | unsigned long micros(void);
      |               ^~~~~~
In file included from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:4,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
src/esphome/core/hal.h:38:10: note: candidate: 'uint32_t esphome::micros()'
   38 | uint32_t micros();
      |          ^~~~~~
In file included from .piolibdeps/living-room-aircon/NeoPixelBus/src/NeoPixelBus.h:102,
                 from src/esphome/components/neopixelbus/neopixelbus_light.h:12,
                 from src/esphome.h:73,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266UartMethod.h: In destructor 'NeoEsp8266AsyncUart<T_UARTFEATURE, T_UARTCONTEXT>::~NeoEsp8266AsyncUart()':
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266UartMethod.h:236:19: error: call of overloaded 'yield()' is ambiguous
  236 |             yield();
      |                   ^
In file included from .piolibdeps/living-room-aircon/ESPAsyncWebServer-esphome/src/ESPAsyncWebServer.h:24,
                 from src/esphome/components/web_server_base/web_server_base.h:11,
                 from src/esphome/components/captive_portal/captive_portal.h:10,
                 from src/esphome.h:19,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
/config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:193:6: note: candidate: 'void yield()'
  193 | void yield(void);
      |      ^~~~~
In file included from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:4,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
src/esphome/core/hal.h:36:6: note: candidate: 'void esphome::yield()'
   36 | void yield();
      |      ^~~~~
In file included from .piolibdeps/living-room-aircon/NeoPixelBus/src/NeoPixelBus.h:102,
                 from src/esphome/components/neopixelbus/neopixelbus_light.h:12,
                 from src/esphome.h:73,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266UartMethod.h: In member function 'void NeoEsp8266AsyncUart<T_UARTFEATURE, T_UARTCONTEXT>::UpdateUart(bool)':
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266UartMethod.h:260:29: error: call of overloaded 'micros()' is ambiguous
  260 |         _startTime = micros();
      |                             ^
In file included from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/PolledTimeout.h:29,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Stream.h:28,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Udp.h:38,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/ESP8266WiFi/src/WiFiUdp.h:25,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/DNSServer/src/DNSServer.h:3,
                 from src/esphome/components/captive_portal/captive_portal.h:6,
                 from src/esphome.h:19,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
/config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/core_esp8266_features.h:103:15: note: candidate: 'long unsigned int micros()'
  103 | unsigned long micros(void);
      |               ^~~~~~
In file included from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:4,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
src/esphome/core/hal.h:38:10: note: candidate: 'uint32_t esphome::micros()'
   38 | uint32_t micros();
      |          ^~~~~~
In file included from .piolibdeps/living-room-aircon/NeoPixelBus/src/NeoPixelBus.h:102,
                 from src/esphome/components/neopixelbus/neopixelbus_light.h:12,
                 from src/esphome.h:73,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266UartMethod.h: In member function 'bool NeoEsp8266UartMethodBase<T_SPEED, T_BASE, T_INVERT>::IsReadyToUpdate() const':
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266UartMethod.h:369:33: error: call of overloaded 'micros()' is ambiguous
  369 |         uint32_t delta = micros() - this->_startTime;
      |                                 ^
In file included from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/PolledTimeout.h:29,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Stream.h:28,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Udp.h:38,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/ESP8266WiFi/src/WiFiUdp.h:25,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/DNSServer/src/DNSServer.h:3,
                 from src/esphome/components/captive_portal/captive_portal.h:6,
                 from src/esphome.h:19,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
/config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/core_esp8266_features.h:103:15: note: candidate: 'long unsigned int micros()'
  103 | unsigned long micros(void);
      |               ^~~~~~
In file included from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:4,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
src/esphome/core/hal.h:38:10: note: candidate: 'uint32_t esphome::micros()'
   38 | uint32_t micros();
      |          ^~~~~~
In file included from .piolibdeps/living-room-aircon/NeoPixelBus/src/NeoPixelBus.h:102,
                 from src/esphome/components/neopixelbus/neopixelbus_light.h:12,
                 from src/esphome.h:73,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266UartMethod.h: In member function 'void NeoEsp8266UartMethodBase<T_SPEED, T_BASE, T_INVERT>::Initialize()':
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266UartMethod.h:381:35: error: call of overloaded 'micros()' is ambiguous
  381 |         this->_startTime = micros() - getPixelTime();
      |                                   ^
In file included from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/PolledTimeout.h:29,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Stream.h:28,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Udp.h:38,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/ESP8266WiFi/src/WiFiUdp.h:25,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/DNSServer/src/DNSServer.h:3,
                 from src/esphome/components/captive_portal/captive_portal.h:6,
                 from src/esphome.h:19,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
/config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/core_esp8266_features.h:103:15: note: candidate: 'long unsigned int micros()'
  103 | unsigned long micros(void);
      |               ^~~~~~
In file included from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:4,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
src/esphome/core/hal.h:38:10: note: candidate: 'uint32_t esphome::micros()'
   38 | uint32_t micros();
      |          ^~~~~~
In file included from .piolibdeps/living-room-aircon/NeoPixelBus/src/NeoPixelBus.h:102,
                 from src/esphome/components/neopixelbus/neopixelbus_light.h:12,
                 from src/esphome.h:73,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266UartMethod.h: In member function 'void NeoEsp8266UartMethodBase<T_SPEED, T_BASE, T_INVERT>::Update(bool)':
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEsp8266UartMethod.h:394:19: error: call of overloaded 'yield()' is ambiguous
  394 |             yield();
      |                   ^
In file included from .piolibdeps/living-room-aircon/ESPAsyncWebServer-esphome/src/ESPAsyncWebServer.h:24,
                 from src/esphome/components/web_server_base/web_server_base.h:11,
                 from src/esphome/components/captive_portal/captive_portal.h:10,
                 from src/esphome.h:19,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
/config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:193:6: note: candidate: 'void yield()'
  193 | void yield(void);
      |      ^~~~~
In file included from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:4,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
src/esphome/core/hal.h:36:6: note: candidate: 'void esphome::yield()'
   36 | void yield();
      |      ^~~~~
In file included from .piolibdeps/living-room-aircon/NeoPixelBus/src/NeoPixelBus.h:103,
                 from src/esphome/components/neopixelbus/neopixelbus_light.h:12,
                 from src/esphome.h:73,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEspBitBangMethod.h: In member function 'bool NeoEspBitBangMethodBase<T_SPEED, T_PINSET>::IsReadyToUpdate() const':
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEspBitBangMethod.h:246:33: error: call of overloaded 'micros()' is ambiguous
  246 |         uint32_t delta = micros() - _endTime;
      |                                 ^
In file included from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/PolledTimeout.h:29,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Stream.h:28,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Udp.h:38,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/ESP8266WiFi/src/WiFiUdp.h:25,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/DNSServer/src/DNSServer.h:3,
                 from src/esphome/components/captive_portal/captive_portal.h:6,
                 from src/esphome.h:19,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
/config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/core_esp8266_features.h:103:15: note: candidate: 'long unsigned int micros()'
  103 | unsigned long micros(void);
      |               ^~~~~~
In file included from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:4,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
src/esphome/core/hal.h:38:10: note: candidate: 'uint32_t esphome::micros()'
   38 | uint32_t micros();
      |          ^~~~~~
In file included from .piolibdeps/living-room-aircon/NeoPixelBus/src/NeoPixelBus.h:103,
                 from src/esphome/components/neopixelbus/neopixelbus_light.h:12,
                 from src/esphome.h:73,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEspBitBangMethod.h: In member function 'void NeoEspBitBangMethodBase<T_SPEED, T_PINSET>::Initialize()':
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEspBitBangMethod.h:255:27: error: call of overloaded 'micros()' is ambiguous
  255 |         _endTime = micros();
      |                           ^
In file included from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/PolledTimeout.h:29,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Stream.h:28,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Udp.h:38,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/ESP8266WiFi/src/WiFiUdp.h:25,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/DNSServer/src/DNSServer.h:3,
                 from src/esphome/components/captive_portal/captive_portal.h:6,
                 from src/esphome.h:19,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
/config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/core_esp8266_features.h:103:15: note: candidate: 'long unsigned int micros()'
  103 | unsigned long micros(void);
      |               ^~~~~~
In file included from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:4,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
src/esphome/core/hal.h:38:10: note: candidate: 'uint32_t esphome::micros()'
   38 | uint32_t micros();
      |          ^~~~~~
In file included from .piolibdeps/living-room-aircon/NeoPixelBus/src/NeoPixelBus.h:103,
                 from src/esphome/components/neopixelbus/neopixelbus_light.h:12,
                 from src/esphome.h:73,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEspBitBangMethod.h: In member function 'void NeoEspBitBangMethodBase<T_SPEED, T_PINSET>::Update(bool)':
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEspBitBangMethod.h:268:19: error: call of overloaded 'yield()' is ambiguous
  268 |             yield(); // allows for system yield if needed
      |                   ^
In file included from .piolibdeps/living-room-aircon/ESPAsyncWebServer-esphome/src/ESPAsyncWebServer.h:24,
                 from src/esphome/components/web_server_base/web_server_base.h:11,
                 from src/esphome/components/captive_portal/captive_portal.h:10,
                 from src/esphome.h:19,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
/config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:193:6: note: candidate: 'void yield()'
  193 | void yield(void);
      |      ^~~~~
In file included from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:4,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
src/esphome/core/hal.h:36:6: note: candidate: 'void esphome::yield()'
   36 | void yield();
      |      ^~~~~
In file included from .piolibdeps/living-room-aircon/NeoPixelBus/src/NeoPixelBus.h:103,
                 from src/esphome/components/neopixelbus/neopixelbus_light.h:12,
                 from src/esphome.h:73,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
.piolibdeps/living-room-aircon/NeoPixelBus/src/internal/NeoEspBitBangMethod.h:290:27: error: call of overloaded 'micros()' is ambiguous
  290 |         _endTime = micros();
      |                           ^
In file included from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/PolledTimeout.h:29,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Stream.h:28,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/Udp.h:38,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/ESP8266WiFi/src/WiFiUdp.h:25,
                 from /config/.esphome/platformio/packages/framework-arduinoespressif8266/libraries/DNSServer/src/DNSServer.h:3,
                 from src/esphome/components/captive_portal/captive_portal.h:6,
                 from src/esphome.h:19,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
/config/.esphome/platformio/packages/framework-arduinoespressif8266/cores/esp8266/core_esp8266_features.h:103:15: note: candidate: 'long unsigned int micros()'
  103 | unsigned long micros(void);
      |               ^~~~~~
In file included from src/esphome/components/light/light_transformer.h:4,
                 from src/esphome/components/light/light_state.h:11,
                 from src/esphome/core/controller.h:11,
                 from src/esphome/components/api/api_server.h:4,
                 from src/esphome/components/api/api_connection.h:6,
                 from src/esphome.h:3,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.h:20,
                 from src/esphome/components/mitsubishi_heatpump/espmhp.cpp:20:
src/esphome/core/hal.h:38:10: note: candidate: 'uint32_t esphome::micros()'
   38 | uint32_t micros();
      |          ^~~~~~
*** [.pioenvs/living-room-aircon/src/esphome/components/mitsubishi_heatpump/espmhp.cpp.o] Error 1`

--- a/components/mitsubishi_heatpump/espmhp.cpp
+++ b/components/mitsubishi_heatpump/espmhp.cpp
@@ -53,6 +53,11 @@ void MitsubishiHeatPump::check_logger_conflict_() {
 #endif
 }
 
+void MitsubishiHeatPump::banner() {
+    ESP_LOGI(TAG, "ESPHome MitsubishiHeatPump version %s",
+            ESPMHP_VERSION);
+}
+
 void MitsubishiHeatPump::update() {
     // This will be called every "update_interval" milliseconds.
     //this->dump_config();

--- a/components/mitsubishi_heatpump/espmhp.h
+++ b/components/mitsubishi_heatpump/espmhp.h
@@ -21,7 +21,6 @@
 #include "esphome/core/preferences.h"
 
 #include "HeatPump.h"
-using namespace esphome;
 
 #ifndef ESPMHP_H
 #define ESPMHP_H
@@ -41,7 +40,7 @@ static const uint8_t ESPMHP_MAX_TEMPERATURE = 31; // degrees C,
 static const float   ESPMHP_TEMPERATURE_STEP = 0.5; // temperature setting step,
                                                     // in degrees C
 
-class MitsubishiHeatPump : public PollingComponent, public climate::Climate {
+class MitsubishiHeatPump : public esphome::PollingComponent, public esphome::climate::Climate {
 
     public:
 
@@ -58,10 +57,7 @@ class MitsubishiHeatPump : public PollingComponent, public climate::Climate {
         );
 
         // Print a banner with library information.
-        void banner() {
-            ESP_LOGI(TAG, "ESPHome MitsubishiHeatPump version %s",
-                    ESPMHP_VERSION);
-        }
+        void banner();
 
         // Set the baud rate. Must be called before setup() to have any effect.
         void set_baud_rate(int);
@@ -82,16 +78,16 @@ class MitsubishiHeatPump : public PollingComponent, public climate::Climate {
         void update() override;
 
         // Configure the climate object with traits that we support.
-        climate::ClimateTraits traits() override;
+        esphome::climate::ClimateTraits traits() override;
 
         // Get a mutable reference to the traits that we support.
-        climate::ClimateTraits& config_traits();
+        esphome::climate::ClimateTraits& config_traits();
 
         // Debugging function to print the object's state.
         void dump_state();
 
         // Handle a request from the user to change settings.
-        void control(const climate::ClimateCall &call) override;
+        void control(const esphome::climate::ClimateCall &call) override;
 
         // Use the temperature from an external sensor. Use
         // set_remote_temp(0) to switch back to the internal sensor.
@@ -102,7 +98,7 @@ class MitsubishiHeatPump : public PollingComponent, public climate::Climate {
         HeatPump* hp;
 
         // The ClimateTraits supported by this HeatPump.
-        climate::ClimateTraits traits_;
+        esphome::climate::ClimateTraits traits_;
 
         // Allow the HeatPump class to use get_hw_serial_
         friend class HeatPump;
@@ -118,16 +114,16 @@ class MitsubishiHeatPump : public PollingComponent, public climate::Climate {
 
         // various prefs to save mode-specific temperatures, akin to how the IR
         // remote works.
-        ESPPreferenceObject cool_storage;
-        ESPPreferenceObject heat_storage;
-        ESPPreferenceObject auto_storage;
+        esphome::ESPPreferenceObject cool_storage;
+        esphome::ESPPreferenceObject heat_storage;
+        esphome::ESPPreferenceObject auto_storage;
 
-        optional<float> cool_setpoint;
-        optional<float> heat_setpoint;
-        optional<float> auto_setpoint;
+        esphome::optional<float> cool_setpoint;
+        esphome::optional<float> heat_setpoint;
+        esphome::optional<float> auto_setpoint;
 
-        static void save(float value, ESPPreferenceObject& storage);
-        static optional<float> load(ESPPreferenceObject& storage);
+        static void save(float value, esphome::ESPPreferenceObject& storage);
+        static esphome::optional<float> load(esphome::ESPPreferenceObject& storage);
 
     private:
         // Retrieve the HardwareSerial pointer from friend and subclasses.

